### PR TITLE
feature/signup page

### LINKED ISSUE #2: UI - Sign Up Page for users
**Is your feature request related to a problem? Please describe.**
As a user, I want to sign up for code mentoring so that I can access their info about meetups, learning paths and get regular updates about their event.

**Describe the solution you'd like**
When user clicks on Sign Up button on the homepage, the sign up page with form should appear

**Design**
https://www.figma.com/file/eyk5tQgLhIpyiORYfWxeUh/Learning-App?node-id=84%3A0

### PR DESCRIPTION
**Pull request by rjtbansal**
Fixes #2 
- Wired up from Home Page
- Error handling based on duplicate emails
Thanks @tristanMatthias  for all your help with this.

### CODE REVIEW

File: src/pages/signup/index.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    const res = await API.signup(values);
+    if (res.data?.createUser) {
+      updateStatus('User Registered Successfully');
+    } else if (res.errors[0].message) {
+      updateStatus('Duplicate Email');
+    } else {
+      console.error('why did this execute');
+    }
+  };
+
```
Comment (tristanMatthias) [inline src/pages/signup/index.tsx:35]: I would update this to use two different states. `success` (boolean) and `error` (string)

Comment (rjtbansal) [reply src/pages/signup/index.tsx]: Done! Let me know what else needs to be improved :)

File: src/pages/signup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const submit = async (values: UserInput) => {
+    const res = await API.signup(values);
+    if (res.data?.createUser) {
+      isSuccessful(true);
+      updateErrorMsg('');
+    } else if (res.errors[0].message) {
+      isSuccessful(false);
+      updateErrorMsg('Duplicate Email');
+    } else {
+      console.error('why did this execute');
```
Comment (tristanMatthias) [inline src/pages/signup/index.tsx:35]: Remove this, or find out reason

File: src/pages/signup/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+  };
+
+  return <main>
+    <header>
+      <img src="/images/logo.svg" alt="Code Mentoring" className="h-12 mx-auto mt-5" />
+    </header>
+    {
+      !success || error === 'Duplicate Email'
```
Comment (tristanMatthias) [inline src/pages/signup/index.tsx:44]: In this case, all you need is `!success`. You don't need to check the error, and we shouldn't compare strings like this (makes code hard to maintain)

File: src/pages/signup/index.tsx (hunk 2 above)
Comment (rjtbansal) [reply src/pages/signup/index.tsx]: Removed 👍

File: src/pages/signup/index.tsx (hunk 3 above)
Comment (rjtbansal) [reply src/pages/signup/index.tsx]: Removed error and overall simplified the way it was being handled 👍

**Pull request merged**